spec/lib/tasks/unpublish_spec.rb

```diff
@@ -0,0 +1,36 @@
+require "rails_helper"
+
+RSpec.describe "unpublish rake tasks", type: :task do
+  describe "unpublish:redirect_finder" do
+    let(:output) { StringIO.new }
+    let(:task) { Rake::Task["unpublish:redirect_finder"] }
+    before { $stdout = output }
+    after { $stdout = STDOUT }
+
+    before(:each) do
+      task.reenable
+      stub_any_publishing_api_unpublish.and_raise(GdsApi::HTTPServerError.new(500))
+    end
+
+    it "returns error message if incorrect finder slug is given" do
+      expect { task.invoke("aaib_testing", "https://service.gov.uk") }.to raise_error("Could not find file: lib/documents/schemas/aaib_testing.json")
+    end
+
+    it "unpublishes finder with redirect" do
+      content_id = MultiJson.load(File.read("lib/documents/schemas/aaib_reports.json"))["content_id"]
+      stub_publishing_api_unpublish(content_id, body: {
+        type: "redirect",
+        alternative_path: "https://service.gov.uk",
+        discard_drafts: true,
+      })
+      task.invoke("aaib_reports", "https://service.gov.uk")
+      expect(output.string).to include("Publishing API response 200")
+      expect(output.string).to include("Finder unpublished")
```

Comment:
It'd be good to assert that the Publishing API actually gets called - not just the text output. Or is that implicitly done by `stub_publishing_api_unpublish` already?

The spec for `publishing_api:publish_finder` includes some assertions – I'm wondering if there's one for 'unpublish' that we can use, too:

https://github.com/alphagov/specialist-publisher/blob/caf19121fa3a82eec4f0ff828b88a075e09fe275/spec/lib/tasks/publishing_api_spec.rb#L62-L64

Reply:
It is implicitly done since in the before block it's stubbing out everything to error, and this test specifically stubs the specific call to return successfully. but i can look for an explicit assert

Reply:
all good - updated the tests with the explicit assert - it reads better now 👍